libs/application/templates/children-residence-change/src/fields/Overview/SignatureModal/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import { useIntl } from 'react-intl'
+import {
+  Box,
+  Text,
+  AlertMessage,
+  Button,
+  ModalBase,
+  SkeletonLoader,
+} from '@island.is/island-ui/core'
+import { signatureModal } from '../../../lib/messages'
+import { FileSignatureStatus } from '../fileSignatureReducer'
+import * as style from './Modal.treat'
+
+interface SignatureModalProps {
+  signatureStatus: FileSignatureStatus
+  onClose: () => void
+  modalOpen: boolean
+  controlCode?: string
+}
+
+const SignatureModal = ({
+  controlCode,
+  signatureStatus,
+  modalOpen,
+  onClose,
+}: SignatureModalProps) => {
+  const { formatMessage } = useIntl()
+  const hasError = [
+    FileSignatureStatus.REQUEST_ERROR,
+    FileSignatureStatus.UPLOAD_ERROR,
+  ].includes(signatureStatus)
+  return (
+    <ModalBase
+      baseId="signatureDialog"
+      className={style.modal}
+      modalLabel={formatMessage(signatureModal.general.title)}
+      isVisible={modalOpen}
+      onVisibilityChange={(visibility: boolean) => {
+        if (!visibility) {
+          onClose()
+        }
+      }}
+      // When there is an error it should not be possible to close the modal
+      hideOnEsc={hasError}
+      // Passing in tabIndex={0} when there is no tabbable element inside the modal
+      tabIndex={!hasError ? 0 : undefined}
+    >
+      <Box
+        className={style.modalContent}
+        boxShadow="large"
+        borderRadius="standard"
+        background="white"
+        paddingX={[3, 3, 5, 12]}
+        paddingY={[3, 3, 4, 10]}
+      >
+        <Text variant="h1" marginBottom={2}>
+          {formatMessage(signatureModal.general.title)}
+        </Text>
+        {(() => {
```

Comment:
I would personally move this into a sub-component to make it clearer and easier to read, instead of having an anonymous function to handle the displaying logic

Reply:
True, was going to do later but decided to do some of the work now :)